src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/repository/AgencyInternalLocationRepository.kt

```diff
@@ -7,25 +7,8 @@ import java.util.Optional
 
 @Repository
 interface AgencyInternalLocationRepository : JpaRepository<AgencyInternalLocation, Long> {
-  fun findAgencyInternalLocationsByAgencyIdAndLocationTypeAndActive(
-    agencyId: String,
-    locationType: String,
-    active: Boolean,
-  ): List<AgencyInternalLocation>
-
-  fun findAgencyInternalLocationsByAgencyIdAndLocationType(
-    agencyId: String,
-    locationType: String,
-  ): List<AgencyInternalLocation>
 
   fun findOneByDescription(description: String): Optional<AgencyInternalLocation>
-  fun findOneByDescriptionAndAgencyId(description: String, agencyId: String): Optional<AgencyInternalLocation>
   fun findByDescriptionAndAgencyId(description: String, agencyId: String): AgencyInternalLocation?
-  fun findOneByLocationId(locationId: Long): Optional<AgencyInternalLocation>
-  fun findByLocationCodeAndAgencyId(locationCode: String, agencyId: String): List<AgencyInternalLocation>
-  fun findByAgencyIdAndLocationTypeAndActiveAndParentLocationIsNull(
-    agencyId: String,
-    locationType: String,
-    active: Boolean,
-  ): List<AgencyInternalLocation>
+  fun findByAgencyIdAndActiveAndLocationCodeInAndParentLocationIsNull(agencyId: String, active: Boolean = true, locationCodes: List<String> = listOf("VISIT", "VISITS")): AgencyInternalLocation?
 }
```

Comment:
The meat - the parent of all visits is defined as:

- A top level location - e.g. no parent
- Have the code either VISIT or VISITS
- Is active

Only 2 prisons do not satisfy this - so will fallback to rooms being created with no parent